cmd/kubeadm/app/cmd/init.go

```diff
@@ -224,13 +233,19 @@ func AddInitOtherFlags(flagSet *flag.FlagSet, cfgPath *string, skipPreFlight, sk
 		criSocket, "cri-socket", "/var/run/dockershim.sock",
 		`Specify the CRI socket to connect to.`,
 	)
+        flagSet.IntVar(
+                verbosityLevel, "verbosity", 0,
+               `Specify the verbosity level of output.`,
+        )
```

Comment:
I wouldn't add new flag here. better add one from glog:
https://github.com/spf13/pflag#supporting-go-flags-when-using-pflag

You can look at kube-proxy as example here.

Reply:
I have confusion with flag.Parse(). kube-proxy doesn't use it currently at least. If I don't use it then pflag doesn't add command line options used by glog package and if I use that like code given below.
```
// NewCmdInit returns "kubeadm init" command.
func NewCmdInit(out io.Writer) *cobra.Command {
	cfg := &kubeadmapiext.MasterConfiguration{}
	legacyscheme.Scheme.Default(cfg)

	var cfgPath string
	var skipPreFlight bool
	var skipTokenPrint bool
	var dryRun bool
	var featureGatesString string
	var criSocket string
	var ignorePreflightErrors []string
        var verbosityLevel int

	cmd := &cobra.Command{
		Use:   "init",
		Short: "Run this command in order to set up the Kubernetes master.",
		Run: func(cmd *cobra.Command, args []string) {
			var err error

                        // Required to include glog package flags with pflag
                	flag.CommandLine.AddGoFlagSet(goflag.CommandLine)

			if cfg.FeatureGates, err = features.NewFeatureGate(&features.InitFeatureGates, featureGatesString); err != nil {
				kubeadmutil.CheckErr(err)
			}

			legacyscheme.Scheme.Default(cfg)
			internalcfg := &kubeadmapi.MasterConfiguration{}
			legacyscheme.Scheme.Convert(cfg, internalcfg, nil)

			ignorePreflightErrorsSet, err := validation.ValidateIgnorePreflightErrors(ignorePreflightErrors, skipPreFlight)
			kubeadmutil.CheckErr(err)

			i, err := NewInit(cfgPath, internalcfg, ignorePreflightErrorsSet, skipTokenPrint, dryRun, criSocket, verbosityLevel)
			kubeadmutil.CheckErr(err)
			kubeadmutil.CheckErr(i.Validate(cmd))
			kubeadmutil.CheckErr(i.Run(out))
		},
	}

	AddInitConfigFlags(cmd.PersistentFlags(), cfg, &featureGatesString)
	AddInitOtherFlags(cmd.PersistentFlags(), &cfgPath, &skipPreFlight, &skipTokenPrint, &dryRun, &criSocket, &ignorePreflightErrors, &verbosityLevel)

        flag.Parse()
	return cmd
}
``` 

It raises following error while building kubeadm.

```
...
...
...
k8s.io/kubernetes/cmd/kubeadm/app/cmd
k8s.io/kubernetes/cmd/kubeadm/app
k8s.io/kubernetes/cmd/kubeadm
go build k8s.io/kubernetes/cmd/kubeadm: /usr/local/go/pkg/tool/linux_amd64/link: signal: killed
!!! [1129 02:30:56] Call tree:
!!! [1129 02:30:56]  1: /home/vbmade2000/go/src/k8s.io/kubernetes/hack/lib/golang.sh:707 kube::golang::build_binaries_for_platform(...)
!!! [1129 02:30:56]  2: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [1129 02:30:57] Call tree:
!!! [1129 02:30:57]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [1129 02:30:57] Call tree:
!!! [1129 02:30:57]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
Makefile:92: recipe for target 'all' failed
make: *** [all] Error 1
```